From 46965bb66e82216086b1ef7fcd5b90cddc2a64fa Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Fri, 28 Aug 2020 09:39:56 +0300
Subject: [PATCH] [SW-11870] preboot: add pll initialization

This patch adds the infrastructure changes required to initialize
plls and their divs based on config provided. This change is reqd
if pll or div frequency change is necessary in specific platform.
e.g. In case of goya2 pldm, uart clk needs to be bumped up to
250Mhz to get prints.

Change-Id: Ieb006dde7e4e1edc490d7b4f06894683ee28b5ee
---

diff --git a/common/pll.c b/common/pll.c
index 46a67f0..39e812f 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -82,6 +82,8 @@
 	}
 }
 
+__weak void hl_pll_init(struct hl_pll_cfg *hl_pll_cfg, int size) {}
+
 __weak u32 hl_pll_get_freq(u32 pll_base, enum hl_pll_div_out div)
 {
 	u32 nr, nf, od, div_sel, factor, offset;
diff --git a/common/pll.h b/common/pll.h
index 03ef1e9..971963d 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -30,6 +30,18 @@
 	HL_PLL_DIV_SIZE,
 };
 
+struct hl_pll_cfg {
+	char    *name;
+	u32	base;
+	u32     nr;
+	u32     nf;
+	u32     od;
+	u32     nb;
+	int     div_factor[HL_PLL_DIV_SIZE];
+	int     div_sel[HL_PLL_DIV_SIZE];
+};
+
+void hl_pll_init(struct hl_pll_cfg *hl_pll_cfg, int size);
 void udelay_init(void);
 u64 hl_pll_timestamp_freq_get(void);
 u32 hl_pll_timestamp_base_get(void);
